### PR TITLE
Improve peribolos around child teams, uppercase, invitations

### PR DESCRIPTION
**Pull request by fejta**
Update github client to:
* Support both Create (cannot send null parent) and Edit team (must send null parent)
* Return a response in dry-run mode for a variety of PUT/PATCH calls
* Create/use privacy/accept constants

Update peribolos to:
* Handle `BenTheElder` and other non-lowercase logins correctly
* Wait for someone to accept an invite to update their membership.
* Create child teams
* Set privacy to closed for nested teams (required by github)
* Allow deleting everything if `maxDelta = 1.0`
* Use role constants

/assign @cblecker @cjwagner @stevekuznetsov @BenTheElder

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -511,6 +518,29 @@ func (c *Client) EditOrg(name string, config Organization) (*Organization, error
 	return &retOrg, nil
 }
 
+func (c *Client) ListOrgInvitations(org string) ([]OrgInvitation, error) {
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/github/client.go]: Done

File: prow/github/types.go [hunk 2]
```diff
@@ -585,14 +585,19 @@ type Content struct {
 	SHA     string `json:"sha"`
 }
 
+const (
+	PrivacySecret = "secret"
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/github/types.go]: Done

File: prow/github/types.go [hunk 3]
```diff
@@ -649,6 +654,11 @@ type TeamMembership struct {
 	Membership
 }
 
+type OrgInvitation struct {
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/github/types.go]: Done

File: prow/cmd/peribolos/main.go [hunk 4]
```diff
@@ -756,6 +780,15 @@ func configureTeam(client editTeamClient, orgName, teamName string, team org.Tea
 		}
 	}
 
+	if team.Privacy != nil && gt.Privacy != string(*team.Privacy) {
+		patch = true
+		gt.Privacy = string(*team.Privacy)
+
+	} else if team.Privacy == nil && (parent != nil || len(team.Children) > 0) && gt.Privacy != "closed" {
```
Comment (cblecker) [inline prow/cmd/peribolos/main.go]: Should we validate the config and error, instead of just overriding it?

File: prow/github/types.go [hunk 5]
```diff
@@ -585,14 +585,21 @@ type Content struct {
...
 // Team is a github organizational team
 type Team struct {
 	ID           int    `json:"id,omitempty"`
 	Name         string `json:"name"`
 	Description  string `json:"description,omitempty"`
 	Privacy      string `json:"privacy,omitempty"`
-	Parent       *Team  `json:"parent,omitempty"` // Only present in responses
-	ParentTeamID *int   `json:"parent_team_id"`   // Only valid in creates/edits
+	Parent       *Team  `json:"parent,omitempty"`         // Only present in responses
+	ParentTeamID *int   `json:"parent_team_id,omitempty"` // Only valid in creates/edits
```
Comment (cblecker) [inline prow/github/types.go]: Interesting.. so sending this on create with a nil value doesn't work?

File: prow/cmd/peribolos/main.go (hunk 4 above)
Comment (cblecker) [reply prow/cmd/peribolos/main.go]: Ohh okay. I misread it the first time. 

Yeah, those are some dumb defaults. 👎

**Pull request merged**